src/System.Text.Primitives/System/Text/InvariantParser_int.cs

```diff
@@ -98,10 +88,18 @@ unsafe public static bool TryParse(byte* utf8Text, int index, int length, out in
             value = 0;
             bytesConsumed = 0;
             bool negative = false;
+            bool signed = false;
 
             if (utf8Text[index] == '-')
             {
                 negative = true;
+                signed = true;
```

Comment:
Actually, I might need some clarification on this part:

```CSharp
if (bytesConsumed == 1 && signed) // if the first character happened to be a '-', we reset the byte counter so logic proceeds as normal.
{
    bytesConsumed = 0;
}
```

It seems like `signed` should only ever be true if the very first character was '+' or '-'. Otherwise, we would have just aborted parsing. Maybe I'm not understanding this part.